src/panel/category/CategoryPanel.jsx

```diff
@@ -130,6 +121,12 @@ export default class CategoryPanel extends React.Component {
       initialLoading: false,
     });
 
+    if (bbox_extended) {
+      // The returned bbox is sure to contain at least one POI.
+      // Extend the current one to include it.
+      window.map.mb.fitBounds(currentBounds.extend(boundsFromFlatArray(contentBbox)));
```

Comment:
Don't we need to take margins into account when using `fitBounds` ?  
(here and also on line 78)

Reply:
Related to this question: is it expected that a query that returned an extended bbox triggers a second request to the API ? Here `fitBounds` will cause a `map_moveend` event, which fetch data again.  
It's probably not an issue in most cases (although the second query is probably redundant), but this may cause flickering if some results are close to the edge of the screen, and margins are not handled properly here.

Reply:
> is it expected that a query that returned an extended bbox triggers a second request to the API ? Here fitBounds will cause a map_moveend event, which fetch data again.

Yes I noticed that too. Do you see a way to prevent this if we keep this behavior of joining the current bounds and the extended ones?

Reply:
@amatissart I adapted and used an existing call that supports padding to do the "fit bounds" operations. https://github.com/QwantResearch/erdapfel/pull/686/commits/09eb1120ee13d87fa7a28e80cee159cc62bff91d

Reply:
It looks good to me.  
You're right that the second query is necessary at least in some cases, the current behavior is probably acceptable :+1: